File: microservices/rooms/src/main/java/kz/placer/hotels/rooms/models/RoomModel.java
package kz.placer.hotels.rooms.models;

import com.sun.istack.NotNull;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.*;


@Entity
@Data
@Table(name = "rooms")
public class RoomModel {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @NotNull
    private Integer id;

    @NotNull
    private int hotelId;

    @NotNull
    private int number;

    @NotNull
    private int floor;

    @NotNull
    private int sleepingPlaces;

    @NotNull
    private String description;

    @NotNull
    private int rooms;

}

File: mobile/busapp/lib/services/models/room_model.dart
class RoomModel {
  final int id;
  final int hotelId;
  final int number;
  final int floor;
  final int sleepingPlaces;
  final String description;
  final int rooms;

  RoomModel(
    this.id,
    this.hotelId,
    this.description,
    this.number,
    this.floor,
    this.sleepingPlaces,
    this.rooms,
  );

  RoomModel.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        hotelId = json['hotelId'],
        number = json['number'],
        floor = json['floor'],
        sleepingPlaces = json['sleepingPlaces'],
        description = json['description'],
        rooms = json['rooms'];

  Map<String, dynamic> toJson() => {
        'id': id,
        'hotelId': hotelId,
        'number': number,
        'floor': floor,
        'sleepingPlaces': sleepingPlaces,
        'description': description,
        'rooms': rooms,
      };
}
